src/test/java/io/kokuwa/maven/helm/junit/MojoExtension.java

```diff
@@ -112,12 +114,24 @@ public Object resolveParameter(ParameterContext parameterContext, ExtensionConte
 			// preconfigure
 
 			mojo.setChartDirectory(new File("src/test/resources/simple")); // set some sane defaults for tests
-			mojo.setHelmExecutableDirectory(new File("src/it")); // avoid download helm
+			mojo.setHelmExecutableDirectory(determineHelmExecutableDirectory());
 			mojo.setHelmVersion("3.12.0"); // avoid github api
 
 			return mojo;
 		} catch (ReflectiveOperationException e) {
 			throw new ParameterResolutionException("Failed to setup mockito.", e);
 		}
 	}
+
+	/**
+	 * Determines which helm executable to use based on the machine's architecture.
+	 * @return location of appropriate helm executable
+	 */
+	public static File determineHelmExecutableDirectory() {
+		if (System.getProperty("os.arch").equals("aarch64")) {
+			return new File("src/it/helm-executables/aarch64-helm/");
+		} else {
+			return new File("src/it/helm-executables/x86_64-helm/");
+		}
+	}
```

Comment:
although the plugin builds fine on an aarch64 machine, if one wishes to debug, the debugger fails because it cannot work with the original helm executable. this was the workaround added to determine the architecture before setting the helm executable.

question: how are the tests able to be run on an aarch64 machine even though the given helm executable is x86?